src/main/java/org/broadinstitute/hellbender/tools/walkers/GenotypeGVCFs.java

```diff
@@ -268,6 +320,56 @@ private VariantContext calculateGenotypes(VariantContext vc){
         return genotypingEngine.calculateGenotypes(vc, model, null);
     }
 
+    /**
+     * Remove INFO field annotations if no alternate alleles
+     *
+     * @param vc    the variant context
+     * @return variant context with the INFO field annotations removed if no alternate alleles
+     */
+    private VariantContext removeInfoAnnotationsIfNoAltAllele(final VariantContext vc)  {
+
+        // If no alt alleles, remove any RankSumTest or RMSAnnotation attribute
+        if ( vc.getAlternateAlleles().isEmpty() ) {
+            final VariantContextBuilder builder = new VariantContextBuilder(vc);
+
+            for ( final String annotation : infoFieldAnnotationKeyNamesToRemove ) {
+                builder.rmAttribute(annotation);
+            }
+            return builder.make();
+        } else {
+            return vc;
+        }
+    }
+
+    /**
+     * Remove NON-REF alleles from the variant context
+     *
+     * @param vc   the variant context
+     * @return variant context with the NON-REF alleles removed if multiallelic or replaced with NO-CALL alleles if biallelic
+     */
+    private VariantContext removeNonRefAlleles(final VariantContext vc) {
+
+        // If NON_REF is the only alt allele, ignore this site
+        final List<Allele> newAlleles = new ArrayList<>();
+        // Only keep alleles that are not NON-REF
+        for ( final Allele allele : vc.getAlleles() ) {
+            if ( !allele.equals(GATKVCFConstants.NON_REF_SYMBOLIC_ALLELE) ) {
+                newAlleles.add(allele);
+            }
+        }
+
+        // If no alt allele, then remove INFO fields that require alt alleles
+        if ( newAlleles.size() == 1 ) {
+            final VariantContextBuilder builder = new VariantContextBuilder(vc).alleles(newAlleles);
+            for ( final String name : infoHeaderAltAllelesLineNames ) {
+                builder.rmAttributes(Arrays.asList(name));
+            }
+            return builder.make();
+        } else {
+            return vc;
```

Comment:
Is it weird that this line has no test coverage?

Reply:
I'm using the same tests GATK3 uses, but it is a bit surprising.